pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -494,8 +493,19 @@ func (pc *PodController) configurePodSecondaryNetwork(pod *corev1.Pod, networkLi
 
 	// Update the Pod's network status annotation
 	if netStatus != nil {
-		if err := netdefutils.SetNetworkStatus(pc.kubeClient, pod, netStatus); err != nil {
-			klog.ErrorS(err, "Pod network status annotation update failed", "Pod", klog.KObj(pod))
+		podItem, err := pc.kubeClient.CoreV1().Pods(pod.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+		if err != nil {
+			klog.ErrorS(err, "Error getting Pod", "Pod", klog.KObj(pod))
+			return err
+		}
+		oldNetworkStatus, err := netdefutils.GetNetworkStatus(podItem)
+		if err == nil {
+			netStatus = append(netStatus, oldNetworkStatus...)
+		} else {
+			klog.ErrorS(err, "Error getting Pod network status annotation", "Pod", klog.KObj(pod))
```

Comment:
Since there is a possibility that the annotation for the primary interface is not updated yet(error when updating), you may still need to update for the secondary interface even without an annotation.